src/main/java/org/gridsuite/modification/server/entities/equipment/modification/SubstationModificationEntity.java

```diff
@@ -0,0 +1,93 @@
+
+    @Column(name = "substationCountryOp")
+    @Enumerated(EnumType.STRING)
+    private OperationType substationCountryOp;
+
+    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
+    @JoinColumn(name = "substation_modification_id")
+    @OrderColumn(name = "insert_position")
+    List<SubstationFreePropertyEntity> properties;
+
+    public SubstationModificationEntity(@NonNull SubstationModificationInfos substationModificationInfos) {
+        super(substationModificationInfos);
+        assignAttributes(substationModificationInfos);
+    }
+
+    @Override
+    public void update(@NonNull ModificationInfos modificationInfos) {
+        super.update(modificationInfos);
+        assignAttributes((SubstationModificationInfos) modificationInfos);
+    }
+
+    private void assignAttributes(SubstationModificationInfos substationModificationInfos) {
+        this.substationCountryValue = substationModificationInfos.getSubstationCountry() != null ? substationModificationInfos.getSubstationCountry().getValue() : null;
+        this.substationCountryOp = substationModificationInfos.getSubstationCountry() != null ? substationModificationInfos.getSubstationCountry().getOp() : null;
+        List<SubstationFreePropertyEntity> newProperties = substationModificationInfos.getProperties() == null ? null :
+                substationModificationInfos.getProperties().stream()
+                        .map(prop -> SubstationFreePropertyEntity.builder()
+                                .name(prop.getName())
+                                .value(prop.getValue())
+                                .deletionMark(prop.isDeletionMark())
+                                .build())
+                        .collect(Collectors.toList());
+        if (this.properties != null) {
+            // update using the same reference with clear/add (to avoid JPA exception)
+            this.properties.clear();
+            if (newProperties != null) {
+                this.properties.addAll(newProperties);
+            }
+        } else {
+            this.properties = newProperties;
+        }
+    }
+
+    @Override
+    public SubstationModificationInfos toModificationInfos() {
+        return toSubstationModificationInfosBuilder().build();
+    }
+
+    private SubstationModificationInfos.SubstationModificationInfosBuilder<?, ?> toSubstationModificationInfosBuilder() {
+        return SubstationModificationInfos
+                .builder()
+                .uuid(getId())
+                .date(getDate())
+                .equipmentId(getEquipmentId())
+                .equipmentName(AttributeModification.toAttributeModification(getEquipmentNameValue(), getEquipmentNameOp()))
+                .substationCountry(AttributeModification.toAttributeModification(getSubstationCountryValue(), getSubstationCountryOp()))
+                .properties(getProperties() == null || getProperties().size() == 0 ? null :
```

Comment:
CollectionUtils.isEmpty(getProperties())

Reply:
yes, good to know